### PR TITLE
Solution HW 2.4

### PR DESCRIPTION
**Pull request by Bogdan-Demianchuk**

### CODE REVIEW

**Comment by Bogdan-Demianchuk** [comment]
Done 

> Separate this blocks to smaller methods. It's a good practice to have some small methods instead of single and giant :)

Done

**Comment by Bogdan-Demianchuk** [comment]
Added some changes

File: src/main/java/core/basesyntax/FieldXY.java [hunk 1]
```diff
@@ -13,7 +13,27 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        if (robot.getX() > toX) {
+            changerDirection(robot, Direction.LEFT);
+        } else {
+            changerDirection(robot, Direction.RIGHT);
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:20]: Can we move this logic inside method?

File: src/main/java/core/basesyntax/FieldXY.java [hunk 2]
```diff
@@ -13,7 +13,27 @@
...
+        } else {
+            changerDirection(robot, Direction.UP);
+        }
+        while (robot.getY() != toY) {
+            robot.stepForward();
+        }
     }
-}
 
+    public static void changerDirection(Robot robot, Direction direction) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:34]:
```suggestion
    public static void changeDirection(Robot robot, Direction direction) {
```

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/FieldXY.java:34]: method's name should be a verb

Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/FieldXY.java]: Done

File: src/main/java/core/basesyntax/FieldXY.java (hunk 1 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/FieldXY.java]: Done

File: src/main/java/core/basesyntax/FieldXY.java [hunk 3]
```diff
@@ -13,7 +13,36 @@
 public class FieldXY {
 
     public void moveRobot(Robot robot, int toX, int toY) {
-        //your code
+        movingByX(robot, toX);
+        movingByY(robot, toY);
+    }
+
+    public static void movingByX(Robot robot, int toXorY) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:20]:
```suggestion
    public static void moveByX(Robot robot, int toXorY) {
```

File: src/main/java/core/basesyntax/FieldXY.java [hunk 4]
```diff
@@ -13,7 +13,36 @@
...
+            turning(robot, Direction.DOWN);
+        } else {
+            turning(robot, Direction.UP);
+        }
+        while (robot.getY() != toXorY) {
+            robot.stepForward();
+        }
+    }
+
+    public static void turning(Robot robot, Direction direction) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:42]:
```suggestion
    public static void turn(Robot robot, Direction direction) {
```

File: src/main/java/core/basesyntax/FieldXY.java [hunk 5]
```diff
@@ -13,7 +13,36 @@
...
+            turning(robot, Direction.LEFT);
+        } else {
+            turning(robot, Direction.RIGHT);
+        }
+        while (robot.getX() != toXorY) {
+            robot.stepForward();
+        }
+    }
+
+    public static void movingByY(Robot robot, int toXorY) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/FieldXY.java:31]: same

Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/FieldXY.java]: Done

File: src/main/java/core/basesyntax/FieldXY.java (hunk 4 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/FieldXY.java]: Done

File: src/main/java/core/basesyntax/FieldXY.java (hunk 3 above)
Comment (Bogdan-Demianchuk) [reply src/main/java/core/basesyntax/FieldXY.java]: Done

**Pull request closed without merging**